### PR TITLE
[AzureMonitor] update OTel to 1.9.0-rc.1

### PR DESCRIPTION
**Pull request by TimothyMothra**

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/MessageData.cs [hunk 1]
```diff
@@ -17,7 +17,10 @@ public MessageData(int version, LogRecord logRecord) : base(version)
             Properties = new ChangeTrackingDictionary<string, string>();
             Measurements = new ChangeTrackingDictionary<string, double>();
             Message = LogsHelper.GetMessageAndSetProperties(logRecord, Properties).Truncate(SchemaConstants.MessageData_Message_MaxLength);
+#pragma warning disable CS0618 // Type or member is obsolete
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/MessageData.cs:20]: Is this document in the changelog of OpenTelemetry SDK?

**Comment by rajkumar-rangaraj** [comment]
Need the removal of "Type or member is obsolete" from all UseAzureMonitor API

https://github.com/Azure/azure-sdk-for-net/blob/40e256208b0b20a0690a4f42ed53394331054045/sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/OpenTelemetryBuilderExtensions.cs#L85

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/MessageData.cs (hunk 1 above)
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/MessageData.cs:20]: I found it in the 1.6.0-alpha changelog. We've never taken a dependency on a pre-release version of OpenTelemetry, so this is the first time we've been exposed to this change.

https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/CHANGELOG.md#160-alpha1

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/MessageData.cs:20]: I still recommend removing this warning if it is not a blocker.

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/MessageData.cs:20]: Leave a TODO to remove this with stable release.

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by lmolkova — APPROVED** [review]

**Review by JoshLove-msft — APPROVED** [review]

File: eng/Packages.Data.props [hunk 2]
```diff
@@ -168,9 +168,8 @@
 
   <ItemGroup Condition="$(MSBuildProjectName.StartsWith('Azure.Monitor.OpenTelemetry'))">
     <!-- OpenTelemetry dependency approved for Azure.Monitor.OpenTelemetry.Exporter package only -->
-    <PackageReference Update="OpenTelemetry" Version="1.8.1" />
-    <PackageReference Update="OpenTelemetry.Exporter.InMemory" Version="1.8.1" />
-    <PackageReference Update="OpenTelemetry.Extensions.Hosting" Version="1.8.1" />
+    <PackageReference Update="OpenTelemetry" Version="1.9.0-rc.1" />
```
Comment (annelo-msft) [inline eng/Packages.Data.props:171]: QQ - is it possible that GA libraries would get a dependency on a non-GA package with this change?

Comment (TimothyMothra) [reply eng/Packages.Data.props:171]: First, this only applies to my Azure.Monitor packages:
https://github.com/Azure/azure-sdk-for-net/blob/759632a80000e7149a81b5a6a26ec4656de405d6/eng/Packages.Data.props#L169-L179

Second, I would expect that if I tried to release a GA version, that the API Review tool would block due to the pre-release packages. So, to answer your question, no. :)
But this assumes we trust the API Review to catch this. :D
If this is unacceptable, I'm open to discussing alternatives.

Comment (annelo-msft) [reply eng/Packages.Data.props:171]: I had missed that it's scoped to a single library.  Thanks for the clarification!

Comment (TimothyMothra) [reply eng/Packages.Data.props:171]: FYI, This PR is onboarding to 1.9.0-rc.1 which we want to test some changes.
OpenTelememetry is planning on releasing 1.9.0 GA either Friday (6/14) or Monday (6/17).

For this specific change, I don't think there's any risk of releasing any version with this RC.1 dependency.

However, in the future if having pre-release versions in the Packages.Data.props is going to raise eyebrows, please let me know! You're actually the second person to ask this exact question :)

**Pull request merged**